### PR TITLE
add append for Datasets in Data objects

### PR DESCRIPTION
**Pull request by bendichter**
change table to use the newly defined append method

needed for https://github.com/NeurodataWithoutBorders/pynwb/issues/1067

### CODE REVIEW

File: src/hdmf/container.py [hunk 1]
```diff
@@ -423,6 +429,11 @@ def extend(self, arg):
             self.data.extend(arg)
         elif isinstance(self.data, np.ndarray):
             self.__data = np.append(self.__data, [arg])
+        elif isinstance(h5py.Dataset):
```
Comment (bendichter) [inline src/hdmf/container.py]:
```suggestion
        elif isinstance(self.data, h5py.Dataset):
```

**Review by oruebel — CHANGES_REQUESTED** [review]
It would be nice to have tests for these new cases for append. Other than that this looks good to me.

**Comment by rly** [comment]
Suggestion: on line 58 of src/hdmf/common/table.py: change `self.data.append` to `self.append`
Otherwise, we will have a similar problem that we had before where `self.data` is an `h5py.Dataset` and `h5py.Dataset` does not have the function `append`.

**Comment by bendichter** [comment]
@oruebel yeah, we should write some tests in hdmf. I just wanted to get this in the queue so the whole team could be on board with these changes.

**Comment by bendichter** [comment]
I added the tests to `test_io_map_data.py` as this was the only place in the tests I found an instantiated `Data`.

**Review by rly — APPROVED** [review]

**Review by ajtritt — APPROVED** [review]

**Comment by bendichter** [comment]
@oruebel is this OK to merge?

**Review by oruebel — APPROVED** [review]
@bendichter looks good. Just FYI, this change introduces an issue in that with this ``Data`` is now dependent on the HDF5 backend. This means., if we add new backends the changes are no longer contained to a new HDMFIO class, but we'd need to update ``Data`` as well. However, this is a non-trivial issue. For now, lets merge this and we'll worry about fixing this as we work on getting the Zarr backend ready for merge.

**Comment by bendichter** [comment]
@oruebel Yes, I agree it would be better to define these methods in a way that is supportable within the different backend modules once we are supporting other backends.

**Pull request merged**